packages/esm-service-queues-app/src/patient-queue-header/patient-queue-header.component.tsx

```diff
@@ -15,16 +15,24 @@ import styles from './patient-queue-header.scss';
 
 const PatientQueueHeader: React.FC<{ title?: string }> = ({ title }) => {
   const { t } = useTranslation();
-  const { queueLocations } = useQueueLocations();
+  const { queueLocations, isLoading, error } = useQueueLocations();
   const userSession = useSession();
   const userLocation = userSession?.sessionLocation?.display;
   const currentQueueLocationName = useSelectedQueueLocationName();
 
-  const handleQueueLocationChange = ({ selectedItem }) => {
+  const handleQueueLocationChange = useCallback((selectedItem) => {
     updateSelectedQueueLocationUuid(selectedItem.id);
     updateSelectedQueueLocationName(selectedItem.name);
     updateSelectedServiceName('All');
-  };
+  }, []);
+
+  if (isLoading) {
+    return <p>Loading...</p>; // Show loading state while fetching data
```

Comment:
We cannot put the whole service queue header in loading state when it's just the dropdown in loading condition.

Reply:
![image](https://github.com/openmrs/openmrs-esm-patient-management/assets/51502471/80c74fa3-02a1-4bc4-a823-8f486c1537fa)

Reply:
Ok now I made like this, when it's loading
```
{isLoading ? (
              <div style={{ width: 150}}>
                <DropdownSkeleton />
              </div>
            ) : (
              <Dropdown
                id="typeOfCare"
                label={currentQueueLocationName ?? t('all', 'All')}
                items={[{ id: 'all', name: t('all', 'All') }, ...queueLocations]}
                itemToString={(item) => (item ? item.name : '')}
                type="inline"
                onChange={handleQueueLocationChange}
              />)}
```

<img width="1438" alt="Screenshot 2024-01-11 at 12 33 22 PM" src="https://github.com/openmrs/openmrs-esm-patient-management/assets/94985341/869d8e3d-df4c-424a-bfbf-1da73d15320d">